c10/macros/Macros.h

```diff
@@ -23,6 +23,18 @@
 
 #include "c10/macros/Export.h"
 
+/**
+ * C10_IGNORE_FLOAT_CAST_OVERFLOW specifies a function which deliberately
+ * casts floating values to integer types without verifying the values fall
+ * within the integer type's dynamic range.
+ */
+
+#if defined(__clang__)
+  #define C10_IGNORE_FLOAT_CAST_OVERFLOW __attribute__((no_sanitize("float-cast-overflow")))
```

Comment:
the other example of this is in https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/Utils.h (__ubsan_ignore_float_divide_by_zero__), we should put these in the same place.

Reply:
(and the same format)

Reply:
SGTM!

Reply:
Wait a moment... that would require including an ATen header in a C10 header, and I believe that's against the rules. I think we're stuck with adding this to C10? I can move the other ubsan macros to C10 and keep their formatting, though.